### ISSUE TITLE
Include user uuid in the OCS API

### ISSUE THREAD

**Issue by C0rby** [owncloud/ocis#3271]
## Is your feature request related to a problem? Please describe.
> Hi, is there any chance we can get the user UUID, which currently only exists in the Graph API (AFAIK), in the OCS API as well? It would make stuff a lot easier in the frontend

## Describe the solution you'd like
The uuid should be obtainable from the OCS API.

## Describe alternatives you've considered
The uuid can be obtained from the graph API.

**Comment by C0rby** [comment]
@JammingBen, can you describe why you want the UUID in the OCS API in addition to the graph API?
What is your workflow where you need the UUID? Just trying to understand the problem, so that we can find the best solution.

**Comment by JammingBen** [comment]
The main issue in Web is that we currently deal with both the OCS and the graph API, especially when working with shares. Two concrete issues:

* When checking the permissions of the current user for a space, we simply check the UUID of this user in the `permissions` object of the space. As we don't have the UUID, we need to query the graph API first. That's okay, but it wouldn't be necessary when the UUID was in included in the OCS API.
* When updating the share for a space (e.g. UserA goes from `viewer` to `manager`), then we need to update the `permissions` object of the space in our store. For this we need the UUID of this user - which we don't know. The only way to deal with this is by fetching the whole space again after the role has been changed.

**Comment by C0rby** [comment]
You mean in the response of this endpoint here, right? `https://ocis.ocis-traefik.latest.owncloud.works/ocs/v1.php/cloud/user`

**Comment by JammingBen** [comment]
Yes, but also the OCS share API:

* `https://ocis.ocis-traefik.latest.owncloud.works/ocs/v2.php/apps/files_sharing/api/v1/sharees`
* `https://ocis.ocis-traefik.latest.owncloud.works/ocs/v1.php/apps/files_sharing/api/v1/shares`

(Web seems to be partly using v1 still... I'd expect that to change in the future)

**Comment by C0rby** [comment]
> When updating the share for a space (e.g. UserA goes from viewer to manager), then we need to update the permissions object of the space in our store. For this we need the UUID of this user - which we don't know. The only way to deal with this is by fetching the whole space again after the role has been changed.

Aren't you sending the username in this case? Can you show me an example request?

**Comment by C0rby** [comment]
> When checking the permissions of the current user for a space, we simply check the UUID of this user in the permissions object of the space. As we don't have the UUID, we need to query the graph API first. That's okay, but it wouldn't be necessary when the UUID was in included in the OCS API.

@micbar, what do you think? Does it make sense to add the UUID to the OCS API or should we find another solution?

**Comment by rhafer** [comment]
I know we're not quite there yet, but wasn't it the idea that we deprecate the ocs/v(1|2).php/cloud/user endpoints in ocis over time in favour of the graph API?

**Comment by JammingBen** [comment]
> Aren't you sending the username in this case? Can you show me an example request?

That's right. However, this is not about the request, but about how we handle the update in Web internally. We store the space similar to how we fetch it, except that we simplify the structure a bit. The space object in Web may look like this:

```
{
  id: 1,
  name: "My Space",
  roles: {
    viewer: ["942a25d4-a450-11ec-b909-0242ac120002"],
    editor: [],
    manager: ["942a25d4-a450-11ec-b909-0242ac120003"]
  },
  ...
}
```

Based in the UUIDs in the `roles` object, we then fetch the user IDs (= usernames) and build the shares tree. This way, we can now update shares by passing the user IDs to the OCS API, like you described. Let's say we update the viewer to be a manager now, hence the UUID in the space object above needs to be moved. But at that point, Web has no mapping of UUIDs and user IDs.

There are ways to overcome this. Currently we fetch the whole space again and re-build the space object again, which now has updated `roles`. We could maybe also somehow include the UUIDs in our internal shares tree. So it's not a "must"-request here, rather a "would make life easier". I don't know about the other clients, I remember the iOS client was asking for this as well.
